src/if_checks.c

```diff
@@ -330,6 +330,13 @@ struct check_result *check_name_used_but_not_required_in_if(const struct
 				flavor = "Role Attribute";
 			} else if (name_is_role(ndata) && look_up_in_decl_map(ndata->name, DECL_ROLE)) {
 				flavor = "Role";
+			} else if (name_is_class(ndata) && look_up_in_decl_map(ndata->name, DECL_CLASS)) {
+				/* Ignore kernel classes */
+				if (!is_userspace_class(ndata->name, ndata->traits)) {
```

Comment:
Shouldn't this just be part of the `name_is_class()` else if clause two lines above so we fall through to the main else, rather than repeating that code here?  It seems more maintainable that way.

Either that or add it to the `system_r` exception at the top.

Reply:
Merged the condition.